service/wallet_service.go

```diff
@@ -42,10 +47,28 @@ type WalletService struct {
 	doneChan chan struct{}
 }
 
+type HashAndHeight struct {
+	Height    uint32    `json:"height"`
+	Hash      string    `json:"string"`
+	Timestamp time.Time `json:"timestamp"`
+}
+
 const nullHash = "0000000000000000000000000000000000000000000000000000000000000000"
 
-func NewWalletService(db wallet.Datastore, km *keys.KeyManager, client client.APIClient, params *chaincfg.Params, coinType wallet.CoinType) *WalletService {
-	return &WalletService{db, km, client, params, coinType, 0, nullHash, []func(wallet.TransactionCallback){}, sync.RWMutex{}, make(chan struct{})}
+func NewWalletService(db wallet.Datastore, km *keys.KeyManager, client client.APIClient, params *chaincfg.Params, coinType wallet.CoinType, repoPath string) (*WalletService, error) {
+	f, err := ioutil.ReadFile(path.Join(repoPath, coinType.String()+".json"))
```

Comment:
This code offers no guarantees that it won't accidentally clobber something else that was accidentally in its way. Express the caching as an interface and use it here:
```
type Cacher interface{
  Get(string) interface{}, error
  Set(string, interface{}) error
}
```